README.md

```diff
@@ -52,102 +31,110 @@ import "github.com/roserocket/xerrs"
 //....
 
 if data, err := MyFunc(); err != nil {
-    err = xerrs.Extend(err) // extend error
+    err := xerrs.Extend(err) // extend error
 
     //....
 
-    xerrs.Stack(err) // print detailed error information including its stack
+    fmt.Println(err.Error())    // prints error
+
+    if x, ok := err.(*xerr); ok {
+        fmt.Println(x.Details()) // print detailed error information including its stack
+    } else {
+        fmt.Println(err.Error()) // print basic error message if it is original error
+    }
+
+    //....
 }
 ```
 
 ### Deferred logging + masking example
 
 ```go
+func ErrDetails(err error) string {
+    if x, ok := err.(*xerr); ok {
+        fmt.Println(x.Details()) // print detailed error information including its stack
```

Comment:
i think this accepts an int.
also this could be a spot to comment that Details prints cause error, mask if specified, and stack